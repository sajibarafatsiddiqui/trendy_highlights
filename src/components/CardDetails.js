import React from 'react';
import { Card, Col, Row } from 'react-bootstrap';
import { PropTypes } from 'prop-types';
import style from 'styles/Details.module.css';

const CardDetails = ({
  startDate, historyRate, today, liveRate,
}) => (
  <Row xs={1} md={1} className={style.details_currency}>
    <Col className="d-flex justify-content-between flex-1">
      <p className="text-start m-3">History Date</p>
      <p className="text-end m-3">{startDate}</p>
    </Col>
    <Col className="d-flex justify-content-between">
      <Card.Text className="text-start m-3">History Rate</Card.Text>
      <p className="text-end m-3">{historyRate}</p>
    </Col>
    <Col className="d-flex justify-content-between">
      <p className="text-start m-3">Current Date</p>
      <p className="text-end m-3">{today}</p>
    </Col>
    <Col className="d-flex justify-content-between">
      <p className="text-start m-3">Latest Rate</p>
      <p className="text-end m-3">{liveRate}</p>
    </Col>
    <Col className="d-flex justify-content-between">
      <p className="text-start m-3">Rate Changed</p>
      <p className="text-end m-3">{(liveRate - historyRate).toFixed(4)}</p>
    </Col>
    <Col>
      {(liveRate - historyRate).toFixed(4) > 0 ? (
        <p className="text-center text-success">
          Rate is increased by &nbsp;
          {(liveRate - historyRate).toFixed(4)}
        </p>
      )
        : (
          <p className="text-center text-danger">
            Rate is decreaded by &nbsp;
            {(-1) * (liveRate - historyRate).toFixed(4)}
          </p>
        )}
    </Col>
  </Row>
);

CardDetails.propTypes = {
  startDate: PropTypes.string.isRequired,
  historyRate: PropTypes.number.isRequired,
  today: PropTypes.string.isRequired,
  liveRate: PropTypes.number.isRequired,
};

export default CardDetails;
