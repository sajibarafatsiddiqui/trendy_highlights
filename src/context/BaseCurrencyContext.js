import React from 'react';

const BaseCurrencyContext = React.createContext();
export default BaseCurrencyContext;
