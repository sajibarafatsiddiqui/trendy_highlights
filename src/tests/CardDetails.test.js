import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from 'redux/store';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import { BrowserRouter as Router } from 'react-router-dom';
import CardDetails from 'components/CardDetails';

describe('Tests for <CardDetails /> component', () => {
  const cardDetails = (
    <CardDetails
      startDate="2023-02-01"
      historyRate={100.23}
      today="2023-03-17"
      liveRate={100.50}
    />
  );
  const baseCurrency = 'USD';

  const wrap = (component) => (
    <Provider store={store}>
      <BaseCurrencyContext.Provider value={{ baseCurrency }}>
        <Router>
          {component}
        </Router>
      </BaseCurrencyContext.Provider>
    </Provider>
  );
  it('Should render the content correctly', () => {
    render(wrap(cardDetails));
    expect(screen.getByText('2023-02-01')).not.toBeNull();
    expect(screen.getByText('100.5')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(cardDetails)).toJSON()).toMatchSnapshot();
  });
});
