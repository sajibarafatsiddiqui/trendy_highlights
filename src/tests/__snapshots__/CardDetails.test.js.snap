// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <CardDetails /> component Should match the snapshot 1`] = `
<div
  className="details_currency row row-cols-md-1 row-cols-1"
>
  <div
    className="d-flex justify-content-between flex-1 col"
  >
    <p
      className="text-start m-3"
    >
      History Date
    </p>
    <p
      className="text-end m-3"
    >
      2023-02-01
    </p>
  </div>
  <div
    className="d-flex justify-content-between col"
  >
    <p
      className="text-start m-3 card-text"
    >
      History Rate
    </p>
    <p
      className="text-end m-3"
    >
      100.23
    </p>
  </div>
  <div
    className="d-flex justify-content-between col"
  >
    <p
      className="text-start m-3"
    >
      Current Date
    </p>
    <p
      className="text-end m-3"
    >
      2023-03-17
    </p>
  </div>
  <div
    className="d-flex justify-content-between col"
  >
    <p
      className="text-start m-3"
    >
      Latest Rate
    </p>
    <p
      className="text-end m-3"
    >
      100.5
    </p>
  </div>
  <div
    className="d-flex justify-content-between col"
  >
    <p
      className="text-start m-3"
    >
      Rate Changed
    </p>
    <p
      className="text-end m-3"
    >
      0.2700
    </p>
  </div>
  <div
    className="col"
  >
    <p
      className="text-center text-success"
    >
      Rate is increased by Â 
      0.2700
    </p>
  </div>
</div>
`;
